src/main/scala/io/gatling/build/compile/GatlingCompilerSettingsPlugin.scala

```diff
@@ -25,28 +25,46 @@ object GatlingCompilerSettingsPlugin extends AutoPlugin {
 
   override def trigger = allRequirements
 
+  trait GatlingCompilerSettingsKeys {
+    val javaReleaseVersion = settingKey[String]("Java release releaseVersion")
+  }
+
+  object GatlingCompilerSettingsKeys extends GatlingCompilerSettingsKeys
+  object autoImport extends GatlingCompilerSettingsKeys
+
+  import autoImport._
+
+  private val DefaultJavaReleaseVersion = "1.8"
   private val isJava8 = scala.util.Properties.javaVersion.startsWith("1.8")
 
-  private val JavacOptions =
-    if (isJava8) {
-      Seq("-source", "1.8", "-target", "1.8")
+  private val JavacOptions = Def.setting {
+    val version = javaReleaseVersion.value
+    if (version == DefaultJavaReleaseVersion) {
+      if (isJava8) {
+        Seq("-source", "1.8", "-target", "1.8")
+      } else {
+        Seq("--release", "1.8")
+      }
     } else {
-      Seq("--release", "8")
+      // When user overwrite java release version, we leave him the responsibility
+      Seq.empty
     }
+  }
 
-  private val ScalacOptions = {
+  private val ScalacOptions = Def.setting {
+    val releaseVersion = javaReleaseVersion.value
     val base =
       Seq(
         "-encoding",
         "UTF-8",
         "-deprecation",
         "-feature",
         "-unchecked",
-        "-language:implicitConversions",
-        "-target:jvm-1.8"
+        "-language:implicitConversions"
       )
 
-    if (isJava8) {
+    // When user overwrite java release version, we leave him the responsibility
+    if (isJava8 || releaseVersion != DefaultJavaReleaseVersion) {
       base
     } else {
       base ++ Seq("-release", "8")
```

Comment:
What's the difference between `-target` and `-release`???

> -release:
> Compile for a specific version of the Java platform. Supported targets: 6, 7, 8, 9

> -target:
> Target platform for object files. ([8],9,10,11,12)

Reply:
A good explanation: https://stackoverflow.com/a/43103038
> javac, --release N is equivalent to -source N -target N -bootclasspath <bootclasspath-from-N>